keps/sig-node/4188-kubelet-pod-readiness-api/README.md

```diff
@@ -217,75 +240,90 @@ the Pods. We don't want this API to return partial data.
-    rpc ListPodStatus(PodStatusListRequest) returns (PodStatusListResponse) {}
-    // WatchPodStatus returns a stream of List of PodStatus
-    // Whenever a pod state change api returns the new list
-    rpc WatchPodStatus(PodStatusWatchRequest) returns (stream PodStatusWatchResponse) {}
-    // GetPodStatus returns a PodStatus for given pod's UID
-    rpc GetPodStatus(PodStatusGetRequest) returns (PodStatusGetResponse) {}
+import "google/protobuf/field_mask.proto";
+
+service Pods {
+    // ListPods returns a list of v1.Pod, filtered by field mask.
+    rpc ListPodStatus(PodListRequest) returns (PodListResponse) {}
+    // WatchPods returns a stream of list of PodInfo, filtered by field mask.
+    // Whenever a pod state changes, api returns the new list.
+    rpc WatchPods(PodWatchRequest) returns (stream PodWatchResponse) {}
+    // GetPod returns a PodInfo for given pod's UID, filtered by field mask.
+    rpc GetPod(PodGetRequest) returns (PodGetResponse) {}
 }
 
-// PodCondition aligns with v1.PodCondition.
-message PodCondition {
-    PodConditionType Type = 1;
-    ConditionStatus Status = 2;
-    Timestamp LastProbeTime = 3;
-    Timestamp LastTransitionTime = 4;
-    string Reason = 5;
-    string Message = 6;
+// PodListRequest allows specifying a field mask.
+message PodListRequest {
+  // Optional field mask in the gRPC metadata.
 }
 
-// PodConditionType aligns with v1.PodConditionType
-enum PodConditionType {
-    ContainersReady = 0;
-    Initialized = 1;
-    Ready = 2;
-    PodScheduled = 3;
-    DisruptionTarget = 4;
+// PodListResponse returns a list of v1.Pod.
+message PodListResponse {
+  repeated v1.Pod pods = 1;
 }
 
-// ConditionStatus aligns with v1.ConditionStatus
-enum ConditionStatus {
-    True = 0;
-    False = 1;
-    Unknown = 2;
+// PodWatchRequest allows specifying a field mask.
+message PodWatchRequest {
+  // Optional field mask in the gRPC metadata.
 }
 
-// PodStatus returns a Pod details and list of status Conditions with deletion info.
-message PodStatus {
-    string podUID = 1;
-    string podNamespace = 2;
-    string podName = 3;
-    bool static = 4;
-    repeated PodCondition conditions = 5;
-    Timestamp DeletionTimestamp = 3;
+// PodWatchResponse returns a v1.Pod, as a stream.
+message PodWatchResponse {
+  v1.Pod pods = 1;
 }
 
-// PodStatusResponse returns a stream of List of PodStatus.
-// Whenever a Pod state changes it will return the new list.
-message PodStatusListResponse {
-    // PodStatus includes the Readiness information of Pods.
-    // In the future it may be extended to include additional information.
-    repeated PodStatus Pods = 1;
+// PodGetRequest contains Pod UID and optional field mask.
+message PodGetRequest {
+  string podUID = 1;
+  // Optional field mask in the gRPC metadata.
 }
 
-// PodStatusGetRequest contains Pods UID
-message PodStatusGetRequest {
-    string podUID = 1;
+// PodGetResponse returns a v1.Pod.
+message PodGetResponse {
+  v1.Pod pod = 1;
```

Comment:
Let's be very explicit in here suggesting that the pod is a full Pod as declared in https://github.com/kubernetes/kubernetes/blob/9343d041209560f487cdf9121d5400a5685fa27a/pkg/apis/core/types.go#L4570 and how the versioning will work.